tests/test/org/commcare/util/test/SqlStorageIndexedTests.java

```diff
@@ -0,0 +1,194 @@
+        l2.setEntry("test_section_id_2", "test_entry_id_2", 2345);
+
+        l3 = new Ledger("ledger_entity_id_3");
+        l3.setID(-1);
+        l3.setEntry("test_section_id_3", "test_entry_id_3", 2345);
+    }
+
+    @Test
+    public void testSqlCaseStorage() {
+        try {
+
+            PrototypeFactory mPrototypeFactory = new PrototypeFactory();
+            mPrototypeFactory.addClass(Case.class);
+
+            String storageKey = "TFCase";
+            String username = "sql-storage-test";
+
+            caseStorage = new SqliteIndexedStorageUtility<Case>(Case.class, username, storageKey);
+
+            caseStorage.write(a);
+
+            Case readCase = caseStorage.read(1);
+            assertEquals("a_case_name", readCase.getName());
+            assertEquals(1, readCase.getID());
+            assertEquals(1, caseStorage.getNumRecords());
+
+            int bID = caseStorage.add(b);
+            readCase = caseStorage.read(bID);
+            assertEquals(bID, readCase.getID());
+            assertEquals("b_case_id", readCase.getCaseId());
+            assertEquals(2, caseStorage.getNumRecords());
+
+            int id = caseStorage.add(c);
+            assertEquals(3, caseStorage.getNumRecords());
+            readCase = caseStorage.getRecordForValue("case-id", "c_case_id");
+            assertEquals(id, readCase.getID());
+            assertEquals(caseStorage.getIDsForValue("case-type", "case_type_ipsum").size(), 3);
+
+            caseStorage.remove(1);
+
+            assertEquals(2, caseStorage.getNumRecords());
+            try {
+                caseStorage.read(1);
+                org.junit.Assert.fail();
+            } catch(NullPointerException e){
+                //good
+            }
+
+            Iterator iterator = caseStorage.iterator();
+            while(iterator.hasNext()){
+                Case mCase = (Case)iterator.next();
+                String caseId = mCase.getCaseId();
+                assertTrue(caseId.equals("b_case_id") || caseId.equals("c_case_id"));
+            }
+
+            caseStorage.removeAll();
+
+            assertEquals(0, caseStorage.getNumRecords());
+
+        }
+        catch (Exception e) {
+            e.printStackTrace();
+            fail("Unexpected exception " + e.getMessage());
+        } finally {
+            SqlSandboxUtils.deleteDatabaseFolder();
+        }
+    }
+
+    @Test
+    public void testSqlLedgerStorage() {
+        try {
+
+            PrototypeFactory mPrototypeFactory = new PrototypeFactory();
+            mPrototypeFactory.addClass(Ledger.class);
+
+            String storageKey = "Ledger";
+            String username = "wspride";
+
+            ledgerStorage = new SqliteIndexedStorageUtility<Ledger>(Ledger.class, username, storageKey);
+
+            ledgerStorage.write(l);
+            ledgerStorage.write(l2);
+            ledgerStorage.write(l3);
+
+            Vector<Object> ids = ledgerStorage.getIDsForValue("entity_id", "ledger_entity_id");
```

Comment:
`getIDsForValue` returns a vector (of integers), so you should remove the `<Object>`. I would leave out the `<Integer>` annotation to prevent lint warnings, but that is just my preference